app/controllers/api/console.php

```diff
@@ -111,3 +115,29 @@
 
         $response->chunk('', true);
     });
+
+App::get('v1/console/resources/:resourceId')
+    ->desc('Check resource ID availability')
+    ->groups(['api', 'projects'])
+    ->label('scope', 'projects.read')
+    ->label('sdk.auth', [APP_AUTH_TYPE_ADMIN])
+    ->label('sdk.namespace', 'console')
+    ->label('sdk.method', 'checkResourceAvailability')
+    ->label('sdk.response.code', Response::STATUS_CODE_OK)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_NONE)
+    ->label('abuse-limit', 10)
+    ->label('abuse-key', 'userId:{userId}')
+    ->param('resourceId', '', new UID(), 'ID of the resource.')
+    ->param('type', '', new WhiteList(['rules']), 'Resource type.')
+    ->inject('response')
+    ->inject('dbForConsole')
+    ->action(function (string $resourceId, string $type, Response $response, Database $dbForConsole) {
+        $document = Authorization::skip(fn () => $dbForConsole->getDocument('rules', $resourceId));
+
+        if ($document && !$document->isEmpty()) {
```

Comment:
getDocument() always returns a document, we just need to check if the document is empty